### PR TITLE
feat: support default email feature via portal config

### LINKED ISSUE #820: 通知邮件配置
看到官方文档（v0.9.0）中提到了发布和回滚的邮件模板样例，猜测apollo是否提供了邮件通知功能，即发布和回滚动作触发邮件通知？
除了创建用户时给用户指定邮箱，没发现哪里可以配置邮件通知或email服务器之类。我搭的0.9.0版本，是不是我的理解有误？
如果已有邮件通知功能，还望告知如何配置或如何实现？

### LINKED ISSUE #3008: 配置发布邮件通知
配置发生变更发送邮件通知，要是能集成到UI上操作，是不是更方便一点？

### LINKED ISSUE #3101: 如何配邮箱？有没有一个在web界面可以配置的地方？我看文档是要编译吗？
如何配邮箱？有没有一个在web界面可以配置的地方？我看文档是要编译吗？

### PR DESCRIPTION
**Pull request by nisiyong**
## What's the purpose of this PR

Apollo already has supported to integration email feature, but it needs developers to implement it by themselves.
For easier use of Apollo, I modified the `DefaultEmailService`, so users could enjoy the email notification feature by just add some configurations at the portal.

Properties users concerned as follows:

- **email.enabled**, set `true` to enable email feature.
- **email.config.host**, the SMTP server address.
- **email.config.user**,  the SMTP auth user.
- **email.config.password**, the SMTP auth password.
- **email.supported.envs**, supported ENVs.

And the properties as follows I think they should have default values unless they want to modify:

- **email.sender**, the default value is same as `email.config.user`
- **email.template.framework** 
```
<html><head><style type="text/css">.table{width:100%;max-width:100%;margin-bottom:20px;border-collapse:collapse;background-color:transparent}td{padding:8px;line-height:1.42857143;vertical-align:top;border:1px solid #ddd;border-top:1px solid #ddd}.table-bordered{border:1px solid #ddd}</style></head><body><h3>发布基本信息</h3><table class="table table-bordered"><tr><td width="10%"><b>AppId</b></td><td width="15%">#{appId}</td><td width="10%"><b>环境</b></td><td width="15%">#{env}</td><td width="10%"><b>集群</b></td><td width="15%">#{clusterName}</td><td width="10%"><b>Namespace</b></td><td width="15%">#{namespaceName}</td></tr><tr><td><b>发布者</b></td><td>#{operator}</td><td><b>发布时间</b></td><td>#{releaseTime}</td><td><b>发布标题</b></td><td>#{releaseTitle}</td><td><b>备注</b></td><td>#{releaseComment}</td></tr></table>#{diffModule}#{rulesModule}<br><a href="#{apollo.portal.address}/config/history.html?#/appid=#{appId}&env=#{env}&clusterName=#{clusterName}&namespaceName=#{namespaceName}&releaseHistoryId=#{releaseHistoryId}">点击查看详细的发布信息</a><br><br>如有Apollo使用问题请先查阅<a href="https://github.com/ctripcorp/apollo/wiki">文档</a>，或直接回复本邮件咨询。</body></html>
```
- **email.template.release.module.diff**
```
<h3>变更的配置</h3>
<table class="table table-bordered">
    <tr>
        <td width="10%"><b>Type</b></td>
        <td width="20%"><b>Key</b></td>
        <td width="35%"><b>Old Value</b></td>
        <td width="35%"><b>New Value</b></td>
    </tr>
    #{diffContent}
</table>
```
- **email.template.rollback.module.diff**
```
<h3>变更的配置</h3>
<table class="table table-bordered">
    <tr>
        <td width="10%"><b>Type</b></td>
        <td width="20%"><b>Key</b></td>
        <td width="35%"><b>回滚前</b></td>
        <td width="35%"><b>回滚后</b></td>
    </tr>
    #{diffContent}
</table>
```
- **email.template.release.module.rules**
```
<h3>灰度规则</h3>
#{rulesContent}
<br>
```
## Which issue(s) this PR fixes:

- Fixes #820
- Fixes #3008 
- Fixes #3101 

## Brief changelog

- add maven dependency `javax.mail` in the portal.
- support some new configs in the portal.
- modify EmailService from the empty implementation to the SMTP implementation. 

Follow this checklist to help us incorporate your contribution quickly and easily:

- [x] Read the [Contributing Guide](https://github.com/ctripcorp/apollo/blob/master/CONTRIBUTING.md) before making this pull request.
- [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
- [x] Write necessary unit tests to verify the code.
- [x] Run `mvn clean test` to make sure this pull request doesn't break anything.

### CODE REVIEW

File: apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/defaultimpl/DefaultEmailService.java [hunk 1]
```diff
@@ -1,12 +1,102 @@
...
-    //do nothing
+  public void send(Email email) {
+    if (!portalConfig.isEmailEnabled()) {
+      return;
+    }
+
+    SMTPTransport t = null;
+    try {
+      Properties prop = System.getProperties();
+      prop.put("mail.smtp.auth", "true");
```
Comment (nobodyiam) [inline apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/defaultimpl/DefaultEmailService.java:39]: Is this mandatory? I noticed the following instruction from the [javadoc](https://javaee.github.io/javamail/docs/api/com/sun/mail/smtp/package-summary.html)

> To use SMTP authentication you'll need to set the mail.smtp.auth property (see below) or provide the SMTP Transport with a username and password when connecting to the SMTP server.

Comment (nisiyong) [reply apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/spi/defaultimpl/DefaultEmailService.java]: Thanks, I just remove this line. The test is fine. Please take a look.

**Review by kezhenxu94 — APPROVED** [review]
one nit, otherwise looks good to me

File: pom.xml [hunk 2]
```diff
@@ -349,13 +350,18 @@
...
+			<dependency>
+				<groupId>com.sun.mail</groupId>
+				<artifactId>javax.mail</artifactId>
+				<version>${javax.mail.version}</version>
+			</dependency>
+			<!-- JDK 11+ -->
+			<dependency>
+					<groupId>org.javassist</groupId>
+					<artifactId>javassist</artifactId>
+					<version>${javassist.version}</version>
```
[the remark this answers is not in the document]
Comment (nisiyong) [reply pom.xml:362]: I have fixed it.

**Review by nobodyiam — APPROVED** [review]
LGTM

**Pull request merged**